documentation/getting-started/getting-started-kotlin.asciidoc

```diff
@@ -0,0 +1,164 @@
+---
+title: Using Vaadin with Kotlin
+order: 600
+layout: page
+---
+
+= Using Vaadin with Kotlin
+
+Vaadin supports the Kotlin programming language through the Karibu-DSL library, which contains various extensions to use Vaadin with Kotlin.
+Perhaps most importantly, the library allows you to create your Vaadin UI in a declarative way using a _domain-specific language_ (DSL).
+These features are enabled by the http://www.vaadinonkotlin.eu/index.html[Vaadin on Kotlin] project.
+
+For a quick introduction to the DSL, to have a form with two text input fields, you define them hierarchically under the `init` method (a block itself) as follows:
+
+----
+@Route("")
+class MyView : VerticalLayout() { <1>
```

Comment:
We need to decide which approach to take: whether to extend the root layout or `KComposite`. `KComposite` is definitely better for reusable components since it doesn't leak the API of the root layout into the component itself; however for views I'm not so sure.

However, in order to stay consistent maybe we should always extend `KComposite`. That is also in line with React recommendation to always use composition over inheritance: https://reactjs.org/docs/composition-vs-inheritance.html

Reply:
Let's go with KComposite, for the sake of consistency:

```
@Route("")
class MyView : KComposite() { <1>
    private val root = ui {
        verticalLayout {
            formLayout { <2>
                textField("Name:") { <3>
                    placeholder = "Last name, first name" <4>
                }
                textField("Age:") {
                    width = "5em"
                }
          }
    }
}
```
```

Reply:
I'd propose to add the following comment to the `private val root = ui {` line:

> The `ui {}` function produces a component which you want to see on the screen.